keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -272,24 +369,35 @@ def call(self, inputs, training=True):
             return inputs
 
     def _augment(self, inputs):
-        image = inputs.get(IMAGES, None)
+        raw_image = inputs.get(IMAGES, None)
+        image = raw_image
         label = inputs.get(LABELS, None)
         bounding_boxes = inputs.get(BOUNDING_BOXES, None)
         keypoints = inputs.get(KEYPOINTS, None)
         segmentation_mask = inputs.get(SEGMENTATION_MASKS, None)
+
+        image_ragged = isinstance(image, tf.RaggedTensor)
+        if image_ragged:
+            image = image.to_tensor()
+
         transformation = self.get_random_transformation(
             image=image,
             label=label,
             bounding_boxes=bounding_boxes,
             keypoints=keypoints,
             segmentation_mask=segmentation_mask,
         )
+
         image = self.augment_image(
             image,
             transformation=transformation,
             bounding_boxes=bounding_boxes,
             label=label,
         )
+
+        if image_ragged and not self.output_dense_images:
```

Comment:
We'll also need to export a `self.force_output_ragged_images` API for `RandomScale` on dense images, etc